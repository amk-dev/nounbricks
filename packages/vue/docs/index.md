---
layout: home
nounbricksHero:
  title: nounbricks
  tagline: A component library for noun assets.
  actions:
    - theme: brand
      text: Get Started
      link: https://github.com/amk-dev/nounbricks
    - theme: alt
      text: View On Github
      link: https://github.com/amk-dev/nounbricks

features:
  - icon: 🎨
    title: Render any noun/noun trait with ease
    details: Easy to use components to add nouns assets into your project
  - icon: 🛠️
    title: Fit artwork to bounds
    details: make the noun/noun traits take the minimum space required with just one prop.
  - icon: 🌈
    title: Custom/No Background
    details: Remove backgrounds or add custom backgrounds with just one prop.
  - icon: ⚡️
    title: Written in TypeScript
    details: Providing you with a powerful developer experience.
  - icon: 🖖
    title: Available For Vue
    details: React support coming soon. Be nounish in your framework of choice.
  - icon: 🖖
    title: Opensource
    details: Its opensource. like all software should be.
---
