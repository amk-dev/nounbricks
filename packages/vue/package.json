{
  "name": "@nounbricks/vue",
  "version": "0.1.0-alpha.0",
  "description": "Vue 3 component library for noun assets",
  "main": "dist/index.js",
  "types": "dist/types/index.d.ts",
  "type": "module",
  "scripts": {
    "test": "vitest run",
    "build": "vite build --mode prod && vue-tsc --declaration --declarationDir dist/types --emitDeclarationOnly",
    "docs:dev": "vitepress dev docs",
    "docs:build": "vitepress build docs",
    "docs:serve": "vitepress serve docs",
    "typecheck": "vue-tsc --noEmit"
  },
  "devDependencies": {
    "@testing-library/vue": "^6.6.1",
    "@vitejs/plugin-vue": "^3.0.0",
    "jsdom": "^20.0.0",
    "typescript": "^4.6.4",
    "vite": "^3.0.0",
    "vitepress": "^1.0.0-alpha.4",
    "vitest": "^0.21.1",
    "vue": "^3.2.37",
    "vue-tsc": "^0.38.4"
  },
  "files": [
    "dist"
  ],
  "author": {
    "name": "Akash K"
  }
}
