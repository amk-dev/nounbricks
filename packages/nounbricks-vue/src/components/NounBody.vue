<template>
  <img v-if="renderType == 'img'" :src="svg" />
  <svg
    v-else
    :width="svgAttributes.width"
    :height="svgAttributes.height"
    :viewBox="svgAttributes.viewBox"
    xmlns="http://www.w3.org/2000/svg"
    shape-rendering="crispEdges"
    v-html="svg"
  ></svg>
</template>

<script setup lang="ts">
import { computed, ComputedRef, withDefaults } from "vue";

import { BodyNameWithoutPrefix, BgColor } from "../data/ImageData";

import { useSvgBuilder, Traits } from "../composables/useSvgBuilder";

const props = withDefaults(
  defineProps<{
    body: BodyNameWithoutPrefix;
    fitToBounds?: boolean;
    bgColor?: BgColor;
    renderType?: "svg" | "img";
  }>(),
  {
    bgColor: undefined,
    renderType: "svg",
  }
);

const traits: ComputedRef<Traits> = computed(() => ({
  body: `body-${props.body}`,
}));

const svgOptions = computed(() => ({
  fitToBounds: props.fitToBounds,
  bgColor: props.bgColor,
  renderType: props.renderType,
}));

const { svg, svgAttributes } = useSvgBuilder(traits, svgOptions);
</script>
