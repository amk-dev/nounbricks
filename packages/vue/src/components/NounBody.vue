<template>
  <svg
    :width="svgAttributes.width"
    :height="svgAttributes.height"
    :viewBox="svgAttributes.viewBox"
    xmlns="http://www.w3.org/2000/svg"
    shape-rendering="crispEdges"
    v-html="svg"
  ></svg>
</template>

<script setup lang="ts">
import { computed, ComputedRef } from "vue";

import { BodyNameWithoutPrefix, BgColor } from "../data/ImageData";

import { useSvgBuilder, Traits } from "../composables/useSvgBuilder";

const props = defineProps<{
  body: BodyNameWithoutPrefix;
  fitToBounds?: boolean;
  bgColor?: BgColor;
}>();

const traits: ComputedRef<Traits> = computed(() => ({
  body: `body-${props.body}`,
}));

const svgOptions = computed(() => ({
  fitToBounds: props.fitToBounds,
  bgColor: props.bgColor,
}));

const { svg, svgAttributes } = useSvgBuilder(traits, svgOptions);
</script>
