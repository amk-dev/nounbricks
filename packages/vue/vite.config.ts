import { defineConfig } from "vite";
import vue from "@vitejs/plugin-vue";

// https://vitejs.dev/config/
export default defineConfig({
  build: {
    lib: {
      entry: "src/index.ts",
      name: "nounbricks",
      fileName: "index",
      formats: ["es", "umd"],
    },
    rollupOptions: {
      external: ["vue"],
    },
  },
  plugins: [vue()],
});
