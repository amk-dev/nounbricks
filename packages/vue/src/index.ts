import FullNoun from "./components/FullNoun.vue";
import NounBody from "./components/NounBody.vue";
import NounHead from "./components/NounHead.vue";
import NounAccessory from "./components/NounAccessory.vue";
import NounGlasses from "./components/NounGlasses.vue";

export { FullNoun, NounBody, NounHead, NounAccessory, NounGlasses };
