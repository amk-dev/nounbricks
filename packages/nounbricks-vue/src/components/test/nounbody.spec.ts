import { describe, it, expect, afterEach } from "vitest";
import { render, cleanup } from "@testing-library/vue";
import NounBody from "../NounBody.vue";

type NounBodyProps = InstanceType<typeof NounBody>["$props"];

afterEach(cleanup);

describe("NounBody", () => {
  it("renders the NounBody with background", () => {
    const props: NounBodyProps = {
      body: "bluegrey",
      bgColor: "d5d7e1",
    };

    const { container } = render(NounBody, {
      props,
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the NounBody without background", () => {
    const props: NounBodyProps = {
      body: "bluegrey",
    };

    const { container } = render(NounBody, {
      props,
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the NounBody with bounds fit", () => {
    const props: NounBodyProps = {
      body: "bluegrey",
      fitToBounds: true,
    };

    const { container } = render(NounBody, {
      props,
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
