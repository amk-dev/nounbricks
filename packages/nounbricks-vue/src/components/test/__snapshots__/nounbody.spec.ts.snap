// Vitest Snapshot v1

exports[`NounBody > renders the NounBody with background 1`] = `
<svg
  height="320"
  shape-rendering="crispEdges"
  viewBox="0 0 320 320"
  width="320"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="#d5d7e1"
    height="100%"
    width="100%"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="210"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="220"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="230"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="240"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="250"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="250"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="260"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="260"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="270"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="270"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="280"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="280"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="290"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="290"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="300"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="300"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="310"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="310"
  />
</svg>
`;

exports[`NounBody > renders the NounBody with bounds fit 1`] = `
<svg
  height="110"
  shape-rendering="crispEdges"
  viewBox="90 210 140 110"
  width="140"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="210"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="220"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="230"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="240"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="250"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="250"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="260"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="260"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="270"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="270"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="280"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="280"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="290"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="290"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="300"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="300"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="310"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="310"
  />
</svg>
`;

exports[`NounBody > renders the NounBody without background 1`] = `
<svg
  height="320"
  shape-rendering="crispEdges"
  viewBox="0 0 320 320"
  width="320"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="210"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="220"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="230"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="140"
    x="90"
    y="240"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="250"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="250"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="260"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="260"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="270"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="270"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="280"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="280"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="290"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="290"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="300"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="300"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="20"
    x="90"
    y="310"
  />
  <rect
    fill="#807f7e"
    height="10"
    width="110"
    x="120"
    y="310"
  />
</svg>
`;
